sdk/helper/error/error.go

```diff
@@ -28,3 +30,19 @@ func FormattedMultiError(err *multierror.Error) error {
 	}
 	return err.ErrorOrNil()
 }
+
+// APIErrIs attempts to coerce err into an UnexpectedResponseError to check
+// its status code. Failing that, it will look for str in the error string.
+// If code==0 it will be ignored, same for str==""
+func APIErrIs(err error, code int, str string) bool {
```

Comment:
Could we add a unit test to cover this? It would help catch any changes in the Nomad API we were not expecting.

Reply:
I don't think with the concrete type `api.UnexpectedResponseError`, no, as I mentioned its construction and fields are all private, so I can't really make one here to use in a test.

but I could turn this into an interface instead, with the (I think quite small) risk that an interface defined here could drift from the real thing.

that would also remove the go mod updates, which yes are due to nomad api package getting used here in this method.

Reply:
how does this look to ya? a4f2564679f053b68110da8a00b43497066e6543

for comparison, the full interface for [`api.UnexpectedResponseError` is here](https://github.com/hashicorp/nomad/blob/v1.6.2/api/error_unexpected_response.go#L28-L39), but we only need `Error()` and `StatusCode()` (and possibly `Unwrap()`, which `errors.As` uses as needed)

I also tested by hand that it works properly against current Nomad release (1.6.2) for both policies and jobs going away.

Reply:
I'm gonna go ahead and merge based on your :heavy_check_mark: but let me know if that ^ doesn't work for ya, and I can fix in a follow-up PR.